xmtp_id/src/scw_verifier/mod.rs

```diff
@@ -0,0 +1,37 @@
+mod chain_rpc_verifier;
+
+use async_trait::async_trait;
+use ethers::{
+    providers::{Http, Provider},
+    types::{BlockNumber, Bytes},
+};
+use thiserror::Error;
+
+use crate::associations::AccountId;
+
+pub use self::chain_rpc_verifier::*;
+
+#[derive(Debug, Error)]
+pub enum VerifierError {
+    #[error("calling smart contract {0}")]
+    Contract(#[from] ethers::contract::ContractError<Provider<Http>>),
+    #[error("unexpected result from ERC-6492 {0}")]
+    UnexpectedERC6492Result(String),
+    #[error(transparent)]
+    FromHex(#[from] hex::FromHexError),
+    #[error(transparent)]
+    Abi(#[from] ethers::abi::Error),
+    #[error(transparent)]
+    Provider(#[from] ethers::providers::ProviderError),
+}
+
+#[async_trait]
```

Comment:
We _are_ using dynamic dispatch right now. There's a potential refactor to use this as a generic, but it's a _big_ diff because the SCW verifier is going to live on the `Client` so I wanted to punt it until this change goes in and decide if it's worth it separately.

Reply:
that make sense, I wasn't sure if that changed based off this PR, it would be a pretty big refactor for little in return